src/test/incremental/simd_intrinsic_ice.rs

```diff
@@ -0,0 +1,17 @@
+#![feature(repr_simd, platform_intrinsics)]
+
+// revisions:rpass1 rpass2
```

Comment:
Maybe it'd make more sense to name this after the issue?

Reply:
How tied is what this fixes to the simd intrinsics? E.g. if we refactor them, they might stop failing here, and a subsequent PR could reintroduce the hash collision.

Reply:
pretty much as long as you keep calling `const_field` it should ICE, and once you stop calling `const_field`, it stops needing to be a query and thus also can't ICE anymore